### PR TITLE
Check domains for self signed certs

### PR DESCRIPTION
**Pull request by zeitounator**
Temptative of implementation for #93

### CODE REVIEW

**Comment by zeitounator** [comment]
Hi @IsmaelMartinez. This is what I came up with to test this from office tommorow. It works from home but did not encounter any self signed cert yet.

If successful tomorrow, I'll try to integrate this in the config and force push my changes here (to get totally rid of the hardoded values for debug)

Tell me if I am on the good track to get this PR accepted in the end or if I have to review my implementation (e.g. include this code somewhere else or change the way I test).

Cheers

**Comment by IsmaelMartinez** [comment]
It looks fine, code can be rearranged at a later stage if required.

For testing sake, I will probably recomend to comment most of the code. Just have the following in the method.

```
event.preventDefault();
callback(true);
```
And maybe console.log the URL.

That should probe of it it the issue and, hopefully, ease your testing. 

Final  implementation should have the URLs check, as you have done. 

For the URLs, you should be able to add a config option for them. Code lives under the config folder. There is a README.md there with a bit of info, but, to resume, we are using yargs and you can probably do it with the type array:

https://github.com/yargs/yargs/blob/master/docs/tricks.md#arrays

Do let me know if you got any questions. Hope helps!

**Comment by zeitounator** [comment]
I was able to get a functioning state !!!! And cherry on the cake, I was even able to completely bypass the entering my login/pass with my kerberos auth turned on in gnome online account settings.

Now I still have to make more tests with different proxy settings to make sure I can provide a solution for most user cases.

I will get back to report here. Thanks.

**Comment by IsmaelMartinez** [comment]
Yeah, well, you could check the certificate fingerprint itself. That should be enough as long as you keep your private keys secure.

**Comment by zeitounator** [comment]
The problem is that Electron is returning a base64 encoded sha256 fingerprint whereas other tools like openssl usually give an hex fingerprint. The easier way for a user would be to get the CA/Self signed certificate in a file, register that file in a config. We would then load the certs, get their fingerprint and compare to the one we got from the request. But threre is no way to do this with the Electron Certificate class. The doc describes the class for the object returned on a certificate-error event but there is no way to instanciate such objects from a file. I've found @chilkat/ck-electron4-linux64 package which as a cert object but it will only provide a sha1 fingerprint (and that would add one more dep). I have to admit I'm a little stuck.

**Comment by IsmaelMartinez** [comment]
Maybe you can just provide the base64 encoded sha256 fingerprint in the config file.

Not perfect but should be ok for version 0.2.0. 

People will need to test/try what values are returned from their certificates, but we can ease that by printing the base64/sha256 fingerprint to the console if fails verification.

That should stop the man in the middle attach vector.

Hope helps

**Comment by zeitounator** [comment]
When are you planning to release 0.2.0 ?

**Comment by IsmaelMartinez** [comment]
No fix date yet. I got a "do not disturb" next week due a aws certification, so either by this Friday or it will need to wait a couple of weeks. 

Happy to leave it for the week of the 29th and sort of have a monthly release train... but no fix date.

**Comment by zeitounator** [comment]
@IsmaelMartinez long time for a small change with lots of different implementation tries but here we are.  Let me know if I need to change anything else. Cheers.

File: app/config/index.js [hunk 1]
```diff
@@ -57,6 +57,11 @@ function argv(configPath) {
 				default: '',
 				describe: 'custom CSS styles file location',
 				type: 'string'
+			},
+			customCACertsFingerprints: {
```
Comment (IsmaelMartinez) [inline app/config/index.js:61]: Can you add that info also in the README.md file in the config folder?

Comment (IsmaelMartinez) [reply app/config/index.js:61]: Other than that, it looks good to go.

Comment (zeitounator) [reply app/config/index.js:61]: My bad.... I added some more details on how to get the expected fingerprints.

**Pull request merged**